### PR TITLE
Add convert type function

### PR DESCRIPTION
**Pull request by shenli**
See: https://dev.mysql.com/doc/refman/5.7/en/cast-functions.html#function_cast

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: parser/parser_test.go [hunk 1]
```diff
@@ -296,4 +300,15 @@ func (s *testParserSuite) TestParser0(c *C) {
 	ok = yyParse(l) == 0
 	c.Assert(ok, Equals, true)
 	c.Assert(len(l.Stmts()), Equals, 2)
+
+	// Testcase for CONVERT(expr,type)
+	src = "SELECT CONVERT('111', SIGNED);"
+	l = NewLexer(src)
+	ok = yyParse(l) == 0
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: here ok not be used, u want to check
`c.Assert(yyParse(l), Equals, 0)`?

**Comment by shenli** [comment]
PTAL @qiuyesuifeng

**Comment by siddontang** [comment]
LGTM

**Pull request merged**